src/devices.jl

```diff
@@ -46,7 +53,7 @@ end
 function totalmem(dev::CuDevice)
     mem_ref = Ref{Csize_t}()
     @apicall(:cuDeviceTotalMem, (Ptr{Csize_t}, CuDevice_t), mem_ref, dev)
-    return mem_ref[]
+    return convert(Int, mem_ref[])
```

Comment:
Why do you need to convert this from unsigned to signed? This will be problematic on 32bit systems if `totalmem` exceeds `typemax(Int32)` (which if my math is right would be more than 2GB)